### ISSUE TITLE
XML output of tr:unzip should include directories

### ISSUE THREAD

**Issue by mkraetke** [transpect/unzip-extension#1]
The output of tr:unzip lists only files. To get the directory names, you have to split the path in the `name` attribute with a regular expression. Unfortunately, the listing below would not show an empty directory in the archive. 

zip listing

```
myzip.zip
|--mydir/
|  |--file.xml
|--emptydir/
```

xml output

``` xml
<c:files xmlns:c="http://www.w3.org/ns/xproc-step"
  xml:base="file:/C:/hurz/archive/">
   <c:file name="mydir/file.xml"/>
</c:files>
```

Therefore I would suggest to represent folders with `c:directory` elements. An XML output would look similar to the output of the `p:directory` step.

``` xml
<c:files xmlns:c="http://www.w3.org/ns/xproc-step"
  xml:base="file:/C:/hurz/archive/">
  <c:directory name="mydir">
     <c:file name="file.xml"/>
   </c:directory>
   <c:directory name="emptydir"/>
</c:files>
```

**Comment by mkraetke** [comment]
Another way to represent empty directories is specified below:

``` xml
<c:files xmlns:c="http://www.w3.org/ns/xproc-step"
  xml:base="file:/C:/hurz/archive/">
   <c:file name="emptydir/"/>
</c:files>
```

This solution would fit better in existing transpect pipelines. Perhaps it's possible to switch between XML list and XML tree view with an XProc option.

**Comment by mkraetke** [comment]
With PR https://github.com/transpect/unzip-extension/pull/4, only empty directories are listed. This might be confusing to some users and you still must parse each `c:file/@name` entry to get all files.
A better way to solve this issue would be a tree representation of directories and files:

``` xml
<c:files xmlns:c="http://www.w3.org/ns/xproc-step"
  xml:base="file:/C:/hurz/archive/">
  <c:directory name="mydir">
     <c:file name="file.xml"/>
   </c:directory>
   <c:directory name="emptydir"/>
</c:files>
```

**Comment by gimsieke** [comment]
This is probably much more complicated to prgram, and it necessitates more changes by people who use it in pipelines.
Can’t we agree on flat directory entries for each directory, whether empty or not?

**Comment by mkraetke** [comment]
It would be easy to generate a flat file list from this tree representation for compatibility reasons but we can agree to use this flat file list. I'll write an individual XProc step that generates the mentioned tree representation from a file list.
Since the step fails for certain zip files to list the contents correctly, I'll leave this issue open. Files for testing where already submitted via email.

**Comment by lwittmar** [comment]
[bar.zip](https://github.com/transpect/unzip-extension/files/292882/bar.zip)

created: zip -r bar.zip my-dir/

zipinfo -- Archive:  bar.zip
Zip file size: 482 bytes, number of entries: 3
drwxr-xr-x  3.0 unx        0 bx stor 16-Jun-01 09:40 my-dir/
-rw-r--r--  3.0 unx        7 tx stor 16-Jun-01 09:39 my-dir/a.txt
-rw-r--r--  3.0 unx        7 tx stor 16-Jun-01 09:40 my-dir/b.txt
3 files, 14 bytes uncompressed, 14 bytes compressed:  0.0%

[foo.zip](https://github.com/transpect/unzip-extension/files/292881/foo.zip)

created: zip foo.zip my-dir/a.txt my-dir/b.txt

zipinfo -- Archive:  foo.zip
Zip file size: 340 bytes, number of entries: 2
-rw-r--r--  3.0 unx        7 tx stor 16-Jun-01 09:39 my-dir/a.txt
-rw-r--r--  3.0 unx        7 tx stor 16-Jun-01 09:40 my-dir/b.txt
2 files, 14 bytes uncompressed, 14 bytes compressed:  0.0%

so the "my-dir" is not an entry in foo.zip

Despite having different contents they create the same file-structure when extratced.

**Comment by gimsieke** [comment]
The `/c:files/@xml:base` attribute seems to be missing now

**Comment by mkraetke** [comment]
This issue was resolved by @gimsieke with commit 03bf82fadb980760d5b7ea5c8c66eefc9f454bf4. The [tr:unzip step](https://github.com/transpect/unzip-extension/blob/master/unzip-declaration.xpl#L28-L31) contains now a secondary output port named `list-with-directories` which provides a nested directory structure.
